ui/hooks/useDisplayName.ts

```diff
@@ -40,12 +42,25 @@ export function useDisplayNames(
     const watchedNftName = watchedNFTNames[index];
     const nft = nfts[index];
 
+    const matchedResolution = domainResolutions?.find(
```

Comment:
Can we encapsulate this logic in a local hook such as `useDomainResolutions` that accepts the name requests and returns just the names, as we do for all the other fallbacks?